hydromt_wflow/workflows/demand.py

```diff
@@ -0,0 +1,367 @@
+):
+    """_summary_.
+
+    _extended_summary_
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        _description_
+    """
+    # Reproject to up or downscale
+    non_iri_scaled = ds.raster.reproject_like(
+        ds_like,
+        method=ds_method,
+    )
+
+    # Focus on population data
+    popu_shape = popu.raster.shape
+    if any([_ex < _ey for _ex, _ey in zip(popu_shape, ds_like.raster.shape)]):
+        popu_scaled = popu.raster.reproject_like(
+            ds_like,
+            method="nearest",
+        )
+        scaling = [_ey / _ex for _ex, _ey in zip(popu_shape, ds_like.raster.shape)]
+        scaling = scaling[0] * scaling[1]
+        popu_scaled = popu_scaled * scaling
+    else:
+        popu_scaled = popu.raster.reproject_like(
+            ds_like,
+            method=popu_method,
+        )
+
+    popu_scaled.name = "Population_scaled"
+
+    # Get transform at half degree resolution
+    transform, width, height = transform_half_degree(
+        ds_like.raster.bounds,
+    )
+
+    # Create dummy dataset from this info
+    dda = full_from_transform(
+        transform=transform,
+        shape=(height, width),
+        crs=4326,
+    )
+
+    # Some info regarding the original ds
+    ds.raster.clip_bbox(ds_like.raster.bounds)
+
+    # Some extra info from ds_like
+
+    # Scale the polulation data based on global glob cells
+    popu_down = popu_scaled.raster.reproject_like(
+        dda,
+        method="sum",
+    )
+    popu_redist = popu_down.raster.reproject_like(
+        popu_scaled,
+        method="nearest",
+    )
+    popu_redist = popu_scaled / popu_redist
+
+    # Setup downscaled non irigation domestic data
+    non_iri_down = non_iri_scaled.raster.reproject_like(
+        dda,
+        method="sum",
+    )
+
+    # Loop through the domestic variables
+    for var in ["dom_gross", "dom_net"]:
+        attrs = non_iri_scaled[var].attrs
+        new = non_iri_down[var].raster.reproject_like(
+            non_iri_scaled,
+            method="nearest",
+        )
+        new = new * popu_redist
+        new = new.fillna(0.0)
+        non_iri_scaled[var] = new
+        non_iri_scaled[var] = non_iri_scaled[var].assign_attrs(attrs)
+
+    return non_iri_scaled, popu_scaled
+
+
+def allocate(
+    ds_like: xr.Dataset,
+    min_area: float | int,
+    admin_bounds: object,
+    basins: xr.Dataset,
+    rivers: xr.Dataset,
+):
+    """_summary_.
+
+    _extended_summary_
+
+    Parameters
+    ----------
+    ds_like : xr.Dataset
+        _description_
+    min_area : float | int
+        _description_
+    admin_bounds : object
+        _description_
+    basins : xr.Dataset
+        _description_
+    rivers : xr.Dataset
+        _description_
+    """
+    # Split based on admin bounds
+    sub_basins = basins.copy()
+    sub_basins["uid"] = range(len(sub_basins))
+
+    if admin_bounds is not None:
+        sub_basins = basins.overlay(
+            admin_bounds,
+            how="union",
```

Comment:
https://geopandas.org/en/stable/docs/user_guide/set_operations.html

I need more information than only those acquired by intersection, discarding by values yield more polygons in this case.

Reply:
Needed to dig a little into this, almost thought I absolutely over-engineered this...

However, This has to to with the crappy nature of admin boundaries at the e.g. the coast, this will clip wflow model cells off which will then be lost when rasterizing it to a map. This whole workaround is to combat that.